src/components/profiles/ProfileProvider.js

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useState } from "react";
+export const ProfileContext = createContext();
+
+export const ProfileProvider = (props) => {
+  const [profiles, setProfiles] = useState([]);
+
+  const getProfiles = () => {
+    return fetch(`http://localhost:8000/profiles`, {
+      headers: {
+        Authorization: `Token ${localStorage.getItem("rare_user_token")}`,
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) => res.json())
+      .then(setProfiles);
+  };
+
+  const updateUserRole = (userId, isStaffUpdate) => {
+    return fetch(`http://localhost:8000/profiles/${userId}/update_role`, {
+      method: "PATCH",
+      headers: {
+        Authorization: `Token ${localStorage.getItem("rare_user_token")}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(isStaffUpdate),
+    }).then(getProfiles);
+  };
+
+  return (
+    <ProfileContext.Provider
+      value={{
+        ProfileContext,
```

Comment:
It probably doesn't matter but I guess the `ProfileContext` does not need to be passed in as a value here?

Reply:
guuudd call i think that was an autofill maybe or i was just gettin strange